zenml/core/steps/split/categorical_ratio_split_step.py

```diff
@@ -85,24 +99,37 @@ def __init__(
              column on which to split.
             split_ratio: A dict mapping { split_name: percentage of categories
                                     in split }.
+            unknown_category_policy: String, indicates how to handle categories
+             in the data that are not present in the supplied category list.
         """
         self.categorical_column = categorical_column
 
         split_map = partition_cat_list(categories, split_ratio)
         lint_split_map(split_map)
         self.split_map = split_map
 
+        if unknown_category_policy in self.split_map:
+            self.unknown_category_policy = unknown_category_policy
+        else:
+            self.unknown_category_policy = constants.SKIP
+
         super().__init__(statistics=statistics,
                          schema=schema,
                          categorical_column=categorical_column,
                          split_ratio=split_ratio,
-                         categories=categories)
+                         categories=categories,
+                         unknown_category_policy=unknown_category_policy)
 
     def partition_fn(self):
         return CategoricalPartitionFn, {
             'split_map': self.split_map,
-            'categorical_column': self.categorical_column
+            'categorical_column': self.categorical_column,
+            'unknown_category_policy': self.unknown_category_policy
         }
 
     def get_split_names(self) -> List[Text]:
-        return list(self.split_map.keys())
+        split_names = list(self.split_map.keys())
```

Comment:
Used twice. Can these steps inherit from each other as they seem to have a lot of the same functionality